packages/reports/addon/consumers/request/sort.ts

```diff
@@ -0,0 +1,146 @@
+
+export default class SortConsumer extends ActionConsumer {
+  @service requestActionDispatcher!: RequestActionDispatcher;
+
+  actions = {
+    /**
+     * @action UPSERT_SORT
+     * @param route - route that has a model that contains a request property
+     * @param columnFragment - The column fragment to update sort for
+     * @param direction - the direction of the new sort
+     */
+    [RequestActions.UPSERT_SORT](
+      this: SortConsumer,
+      route: Route,
+      columnFragment: ColumnFragment,
+      direction: SortDirection
+    ) {
+      const { routeName } = route;
+      const { request } = route.modelFor(routeName) as ReportModel;
+      const sort = request.sorts.find(sort => sort.canonicalName === columnFragment.canonicalName);
+
+      if (sort) {
+        set(sort, 'direction', direction);
+      } else {
+        request.addSort({
+          type: columnFragment.type,
+          field: columnFragment.field,
+          parameters: columnFragment.parameters,
+          source: columnFragment.source,
+          direction
+        });
+      }
+    },
+
+    /**
+     * @action REMOVE_SORT
+     * @param route - route that has a model that contains a request property
+     * @param columnFragment - The column fragment to update sort for
+     */
+    [RequestActions.REMOVE_SORT](this: SortConsumer, route: Route, columnFragment: ColumnFragment) {
+      const { routeName } = route;
+      const { request } = route.modelFor(routeName) as ReportModel;
+      const sort = request.sorts.find(sort => sort.canonicalName === columnFragment.canonicalName);
+      if (sort) {
+        request.removeSort(sort);
+      }
+    },
+
+    /**
+     * @action REMOVE_COLUMN
+     * @param route - route that has a model that contains a request property
+     * @param columnMetadata - metadata model of column to remove
+     */
+    [RequestActions.REMOVE_COLUMN](this: SortConsumer, route: Route, columnMetadata: ColumnMetadataModels) {
+      const { routeName } = route;
+      const { request } = route.modelFor(routeName) as ReportModel;
+      const sorts = request.sorts.filter(sort => sort.columnMetadata === columnMetadata);
+
+      // Find and remove any `sorts` for the given column metadata
+      sorts.forEach(sort => {
+        this.requestActionDispatcher.dispatch(RequestActions.REMOVE_SORT, route, sort);
+      });
+    },
+
+    /**
+     * @action REMOVE_COLUMN_WITH_PARAMS
+     * @param route - route that has a model that contains a request property
+     * @param columnMetadata - metadata model of column to remove
+     * @param parameters - metric parameters
+     */
+    [RequestActions.REMOVE_COLUMN_WITH_PARAMS](
+      this: SortConsumer,
+      route: Route,
+      columnMetadata: ColumnMetadataModels,
+      parameters: ColumnFragment['parameters']
+    ) {
+      const { routeName } = route;
+      const { request } = route.modelFor(routeName) as ReportModel;
+      const sorts = request.sorts.filter(
```

Comment:
Could we have more than one? Also, we use cname to find a column above, any reason not to do the same here?

Reply:
We don't have a column fragment here, so I could call canonicalizeMetric but either should work. I'm expecting that sorts will always have a length of 0 or 1, but it seemed right to use `.filter` instead of `.find`

Reply:
> but it seemed right to use .filter instead of .find
Agreeded, should we do the same for the other actions that use `find`?